### PR TITLE
Fix `Unable to get file size` when dataset has no protocol attribute

### PR DESCRIPTION
**Pull request by SajidAlamQB**
## Description
Related to: https://github.com/kedro-org/kedro-viz/issues/1893

User reported getting `Unable to get file size. Object has no attribute '_protocol'` for `DeltaTableDataset`. This PR will aim to resolve it.

## Development notes

Stopped relying on methods from kedro that rely on private attribute, use `fsspec` to find file size.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by ravi-kumar-pilla — APPROVED** [review]
LGTM 💯 ... thank you @SajidAlamQB

File: RELEASE.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ Please follow the established format:
 - Display full dataset type with library prefix in metadata panel (#2136)
 - Enable SQLite WAL mode for Azure ML to fix database locking issues (#2131)
 - Replace `flake8`, `isort`, `pylint` and `black` by `ruff` (#2149)
+- Refactor `DatasetStatsHook` to avoid accessing private dataset attributes (#2174)
```
Comment (astrojuanlu) [inline RELEASE.md:24]: Rather, "avoid showing error when dataset doesn't have file size info"?

**Review by astrojuanlu — COMMENTED** [review]
With 1 caveat LGTM!

**Review by rashidakanchwala — APPROVED** [review]
Thanks @SajidAlamQB

**Pull request merged**